ostd/src/mm/frame/allocator.rs

```diff
@@ -8,17 +8,19 @@ use align_ext::AlignExt;
 
 use super::{meta::AnyFrameMeta, segment::Segment, Frame};
 use crate::{
+    arch::mm::PagingConsts,
     boot::memory_region::MemoryRegionType,
     error::Error,
     impl_frame_meta_for,
-    mm::{paddr_to_vaddr, Paddr, PAGE_SIZE},
+    mm::{paddr_to_vaddr, page_size, Paddr, PAGE_SIZE},
     prelude::*,
     util::ops::range_difference,
 };
 
 /// Options for allocating physical memory frames.
 pub struct FrameAllocOptions {
     zeroed: bool,
+    level: u8,
```

Comment:
let's use a name other than level, level can correspond to execution level in aarch64 which can make this confusing.
Also we need a comment to describe what direction numbers go in here.

Reply:
Does this also effect the `LEVEL` elsewhere? (I don't actually fully understand what is going on here, but I don't need to.)

Reply:
@tewaro I'd prefer to use level to be consistent with asterinas' terminology in other places. Maybe we can consider a broader refactor later?

@arthurp not yet, but it will later. LEVEL is a compile time const used to create caches for different allocation sizes. As we add huge page support, I'll be adding a huge page cache that we will only check if layout.level matches.